vllm/sequence.py

```diff
@@ -320,6 +336,25 @@ def fork(self, new_seq_id: int) -> "Sequence":
         new_seq.seq_id = new_seq_id
         return new_seq
 
+    def get_num_new_tokens(self) -> int:
+        """Get the number of new tokens to be computed.
+
+        Args:
+            remainig_token_budget: The remaining token budgets.
+        Returns:
+            The new number of tokens to be computed. I.e., 1 for decode, prompt
+            size for prefill. If there's not enough remainig_token_budget, it
+            can return the chunked number of new tokens.
+        """
+        num_uncomputed_tokens = self.data.get_num_uncomputed_tokens()
+        if self.data.stage == SequenceStage.DECODE:
+            assert num_uncomputed_tokens == 1
```

Comment:
where is this set? `get_num_uncomputed_tokens` in `RequestMetrics` only returns `return self.get_len() - self.get_num_computed_tokens()`

Reply:
Hmm yeah, this assumes that a new output token is added to sequence (so that get_len is incremented by 1) before update happens. I just removed the assert here